src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultProjectWorkspaceStateGenerator.cs

```diff
@@ -210,29 +214,5 @@ private void OnBackgroundWorkCompleted()
                 NotifyBackgroundWorkCompleted.Set();
             }
         }
-
-        // Internal for testing
-        internal class UpdateItem
```

Comment:
In this situation I wouldn't share this class because each class has their own variant of the pattern that uses it. Meaning, if you start sharing it with the `WorkspaceProjectStateChangeDetector.cs` any change to one might break the other.

Reply:
Makes sense, reverted this, and added a copy into the ChangeDetector